test/module/irohad/torii/processor/transaction_processor_test.cpp

```diff
@@ -140,6 +142,66 @@ TEST_F(TransactionProcessorTest, TransactionProcessorOnProposalTest) {
   validateStatuses<shared_model::interface::StatelessValidTxResponse>(txs);
 }
 
+/**
+ * @given transactions from the same batch
+ * @when transactions sequence is created and propagated @and all transactions
+ * were returned by pcs in proposal notifier
+ * @then all transactions have stateless valid status
+ */
+TEST_F(TransactionProcessorTest, TransactionProcessorOnProposalBatchTest) {
+  using namespace shared_model::validation;
+  using TxValidator =
+      TransactionValidator<FieldValidator,
+                           CommandValidatorVisitor<FieldValidator>>;
+
+  using TxsValidator =
+      UnsignedTransactionsCollectionValidator<TxValidator, BatchOrderValidator>;
+
+  auto transactions =
+      framework::batch::createValidBatch(proposal_size).transactions();
+
+  auto wrapper =
+      make_test_subscriber<CallExact>(tp->transactionNotifier(), proposal_size);
+  wrapper.subscribe([this](auto response) {
+    status_map[response->transactionHash()] = response;
+  });
+
+  auto transaction_sequence_result =
+      shared_model::interface::TransactionSequence::createTransactionSequence(
+          transactions, TxsValidator());
+  auto transaction_sequence =
+      framework::expected::val(transaction_sequence_result).value().value;
+
+  EXPECT_CALL(*mp, propagateTransactionImpl(_)).Times(0);
+  EXPECT_CALL(*pcs, propagate_batch(_))
+      .Times(transaction_sequence.batches().size());
+
+  tp->transactionSequenceHandle(transaction_sequence);
+
+  // create proposal from sequence transactions and notify about it
+  std::vector<shared_model::proto::Transaction> proto_transactions;
+
+  std::transform(
+      transactions.begin(),
+      transactions.end(),
+      std::back_inserter(proto_transactions),
+      [](const auto tx) {
+        return *std::static_pointer_cast<shared_model::proto::Transaction>(tx);
+      });
+
+  auto proposal = std::make_shared<shared_model::proto::Proposal>(
+      TestProposalBuilder().transactions(proto_transactions).build());
+
+  prop_notifier.get_subscriber().on_next(proposal);
+  prop_notifier.get_subscriber().on_completed();
+
+  ASSERT_TRUE(wrapper.validate());
+
+  SCOPED_TRACE("Stateless valid status verification");
+  validateStatuses<shared_model::interface::StatelessValidTxResponse>(
+      proto_transactions);
+}
```

Comment:
Is it possible to write another test, which will check fail case, so that transactions in batch will have STATELESS_INVALID status, if one of them fails?

Reply:
No, the case you explained should be written in torii service test. Here we check that we have created proposal from all transactions from the batch and since they are in proposal, they have passed stateless validation